nxdrive/direct_edit.py

```diff
@@ -308,18 +308,35 @@ def _download(
 
         if not pair:
             if url:
+                max_retry = 3
+                tmp_file = None
                 try:
-                    engine.remote.download(
-                        quote(url, safe="/:"),
-                        file_path,
-                        file_out,
-                        blob.digest,
-                        callback=self.stop_client,
-                        is_direct_edit=True,
-                        engine_uid=engine.uid,
-                    )
+                    for try_count in range(max_retry):
+                        try:
+                            tmp_file = engine.remote.download(
+                                quote(url, safe="/:"),
+                                file_path,
+                                file_out,
+                                blob.digest,
+                                callback=self.stop_client,
+                                is_direct_edit=True,
+                                engine_uid=engine.uid,
+                            )
+                            break
+                        except CorruptedFile:
+                            description = "Unable to perform DirectEdit because of corrupted download\nRetrying..."
+                            log.warning(description, exc_info=True)
```

Comment:
Are they needed? The notification below will be automatically logged: https://github.com/nuxeo/nuxeo-drive/blob/32f4470cd38611c29273ced1a68511d963dd35df/nxdrive/notification.py#L175
A small improvement we could do later is to log the appropriate log level based on the notification level.